projects/igniteui-angular/src/lib/select/select-positioning-strategy.ts

```diff
@@ -31,201 +25,231 @@ export class SelectPositioningStrategy extends ConnectedPositioningStrategy impl
+        Math.round(selectFit.itemRect.top * 100) / 100 <=
+        (selectFit.dropDownList.getBoundingClientRect().bottom * 100) / 100 &&
+        Math.round(selectFit.itemRect.bottom * 100) / 100 >=
+        (selectFit.dropDownList.getBoundingClientRect().bottom * 100) / 100;
+    }
+
+    private manageScrollToItem(selectFit: SelectFit) {
+        // Scroll and compensate the item's container position, when the selected item is not visible.
+        if (this.itemIsInvisible(selectFit)) {
+            const compensation = this.scrollToItem(selectFit);
+            this.compensateYScroll(selectFit, compensation);
+        }
+    }
+
+    private scrollToItem(selectFit: SelectFit): number {
+        const itemPosition = this.calculateScrollPosition(selectFit);
+        if (isIE()) {
+            setTimeout(() => {
+                selectFit.dropDownList.scrollTop = (itemPosition);
+            }, 1);
         } else {
-            return null;
+            selectFit.dropDownList.scrollTop = (itemPosition);
         }
-        return returnVals;
+        return itemPosition;
     }
 
-    position(contentElement: HTMLElement, size: Size, document?: Document, initialCall?: boolean): void {
-        const inputElement = this.select.input.nativeElement;
-        const inputRect = inputElement.getBoundingClientRect() as DOMRect;
-        this.listContainerBoundRect = contentElement.getBoundingClientRect() as DOMRect;
-        const LIST_HEIGHT = this.listContainerBoundRect.height;
-        if (!initialCall) {
-            this.deltaX = inputRect.left - this.itemTextPadding - this.itemTextIndent;
-            const point = new Point(this.deltaX, inputRect.top + this.deltaY);
-            this.settings.target = point;
-            super.position(contentElement, size);
-            return;
+    private calculateScrollPosition(selectFit: SelectFit): number {
+        if (!selectFit.itemElement) { // this check may not be necessary
+            return 0;
         }
 
-        const START = {
-            X: inputRect.left,
-            Y: inputRect.top
-        };
+        const elementRect = selectFit.itemRect;
+        const parentRect = selectFit.dropDownList.getBoundingClientRect();
+        const scrollPosition = elementRect.bottom - parentRect.bottom;
+        return Math.floor(scrollPosition);
+    }
+
+    // This method can be scrambled and combined in manageScrollToItem()
+    private compensateYScroll(selectFit: SelectFit, compensation: number ) {
+        selectFit.verticalOffset += compensation;
+        this.global_yOffset = selectFit.verticalOffset;
+    }
+
+    // Position the items outer container Below or Above the input.
+    fitInViewport(contentElement: HTMLElement, selectFit: SelectFit) {
+        // Position Select component's container below target/input as preferred positioning over above target/input
+        if (this.canFitBelowInput(selectFit) || !this.canFitAboveInput(selectFit)) {
```

Comment:
`canFitBelowInput` and `canFitAboveInput` are both single line methods. Why not calculate both in boolean variable like this

```
        const canFitBelowInput = selectFit.targetRect.top - selectFit.styles.itemTextToInputTextDiff + selectFit.contentElementRect.height <
            selectFit.viewPortRect.bottom;
        const canFitAboveInput = selectFit.targetRect.bottom + selectFit.styles.itemTextToInputTextDiff -
            selectFit.contentElementRect.height > selectFit.viewPortRect.top;
        // Position Select component's container below target/input as preferred positioning over above target/input
        if (canFitBelowInput || !canFitAboveInput) {
       
```

Reply:
done